### DISCUSSION TITLE
Assembling of MPO files

### DISCUSSION THREAD

**Discussion by andsimakov** [python-pillow/Pillow#6373] [Q&A] [answered]
Is it possible to assemble MPO file with Pillow from two source files? The idea is to have one color file as a basic photo and the second one — its adjusted monochrome copy as HDR Gain Map to store it in the MPImage2 tag of the basic file.

it's the way how it works in iOS to display a photo on an OLED display with a wider brightness range. I'd like to make such kind of photos on my own from my Canon MILC files.

**Comment by radarhere** [comment, marked as answer]
There isn't currently a Pillow API for this, no.

https://github.com/python-pillow/Pillow/blob/765d66c069fb35442a01457955c15a8db898f734/src/PIL/MpoImagePlugin.py#L28-L29

**Reply by radarhere** [reply]
I've created PR #6444 to add this functionality.

**Reply by radarhere** [reply]
That PR has now been merged, so you will be able to combine two images into one MPO in the next version of Pillow, scheduled for October 15.

If you would like this functionality before then, you can download Pillow from our main branch and install it from source.

**Reply by andsimakov** [reply]
Thank you very much. I'll try to create MPO files then.
